pkg/foundation/database/database.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:generate mockgen -destination=mock/database.go -package=mock -mock_names=DB=DB . DB
+//go:generate mockgen -typed -destination=mock/database.go -package=mock -mock_names=DB=DB . DB
```

Comment:
I changed all `mockgen` invocations to include the `-typed` flag introduced in mockgen. This makes our tests a bit safer and easier to write, since autocomplete helps you out now.

This is also where the bulk of the additional code comes from. Without this change we would lose quite a bit of lines because of the removal of `pkg/record` and the simplification of the plugin packages.

Reply:
Was there a specific reason why we didn't use typed mocks before?

Reply:
This was added in the `uber-go/mock` fork of the library relatively recently (september last year). We should use it going forward.